### ISSUE TITLE
Subjects with no courses appear when using `Only Show Courses in Session`

### ISSUE THREAD

**Issue by jakeale** [VikeLabs/courseup#194]
**Describe the bug**

When using the `Only Show Courses in Session` filter, subjects with no courses in that session still appear in the subjects list.

**To Reproduce**

1. Go to https://courseup.vikelabs.ca/calendar/202105/
2. Click on the `Only Show Courses in Session` button
3. Click on ACAN, for instance

**Expected behavior**

With the filter enabled, subjects with no courses in the current session should not appear.

**Screenshots**

![UIRumZgEva](https://user-images.githubusercontent.com/25995184/120959637-1c68d000-c70f-11eb-868a-2cf4e06cf721.png)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Microsoft Edge
 - Version [e.g. 22] Latest

**Additional context**

This may not be technically a bug but instead just the result of how the filter functions (since it's not an `Only Show Subjects in Session` filter), and this could instead be labeled as a possible QOL enhancement. If this is the case I can update the issue.

**Comment by aomi** [comment]
@jakeale Thanks for the report. I think you're right on that this isn't a bug but more a "enhancement". 
Given the report I think there's some confusion with the feature, which isn't very good UX and should be addressed.

My concern to hide subjects was it might cause confusion with missing subjects.
I think there's two approaches we could take to tackle this.
1. "Disable" subjects that have no courses in them (making them unclickable )
2. Hide the subject with 0 courses in session.

I'd love to hear any thoughts you might have on the possible enhancement. 

cc: @keithradford @zthh626

**Comment by keithradford** [comment]
Maybe disable the card + have a title that appears when you hover saying `No courses for ADMN in Summer 2021` might work. Prevents the user from going into a subject w/out any classes but provides clarification to why they can't in the first place too.

**Comment by jakeale** [comment]
I agree that hiding subjects altogether might be confusing, I like @keithradford's idea.
